### PR TITLE
Maven integration

### PR DESCRIPTION
**Pull request by crushcoder**
Added a new page for filling the scm-source.json file via Maven and linked it.

### CODE REVIEW

**Comment by hjacobs** [comment]
I'm not very fond of this PR as it's a very specific use case --- a link to an existing "SCM Source Maven Plug In" would be fine, but adding a whole page about custom `pom.xml` configuration?

**Comment by crushcoder** [comment]
...but having no documentation for the probably most used technology and wait for someone to write and maintain a plugin instead?

**Comment by hjacobs** [comment]
@crushcoder we are also using Maven in our team, but we always generate the `scm-source.json` with the `scm-source` Python tool :smirk: You only have to install it once in your CI/CD, or better use a "toolchain" for it (works on STUPS2Go and CloudLobster Jenkins):

```
/tools/run registry.opensource.zalan.do/stups/toolchain-stups:17 -- scm-source -f target/scm-source.json
```

File: user-guide/maven.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+To generate the scm-source with Maven you need a template file which will be filled with properties from a plugin.
+
+``src/main/docker/scm-source.json``:
+
+.. code-block:: src/main/docker/scm-source.json
+
+    { "url": "${project.scm.url}",
+      "revision": "${buildNumber}",
+      "author": "${user.name}",
+      "status": "${scmBranch}" }
```
Comment (hjacobs) [inline user-guide/maven.rst]: the "status" field does not contain the branch, but the `git status` output

**Comment by crushcoder** [comment]
Part of my toolchain is Maven, that's why i didn't use the scripts .  ;-)
I also need that while Maven builds because afterwards Maven builds the docker image.

I was underwhelmed that the documentation is missing this when i started with stups.

**Pull request closed without merging**